rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -29,6 +33,10 @@
   private final TaskScheduler taskScheduler;
   private final ErrorHandler errorHandler;
 
+  // This list is used to retain error adapters in a non-weak reference associated with this use
+  // case handler
+  private final List<OnErrorCallback> errorCallbacks = new ArrayList<>();
```

Comment:
Yeap, so this is not an issue. You are right! As the view is a weak reference this is not going to be a problem 😄 So once we fix my other comment we are ready to merge! Thanks for your help guys.